toolchest_client/tools/tool.py

```diff
@@ -84,6 +86,11 @@ def __init__(self, tool_name, tool_version, tool_args,
         self.custom_docker_image_id = custom_docker_image_id
         self.instance_type = instance_type
         self.volume_size = volume_size
+        # auto-disable streaming if job is async
+        self.streaming_enabled = False if self.is_async else streaming_enabled
+        self.streaming_client = None
+        self.streaming_task = None
```

Comment:
What is the difference between `streaming_client` and `streaming_task`?

Reply:
`streaming_client` is the `StreamingClient` object, which contains the streaming parameters and websocket client.

`streaming_task` is an `asyncio` task that runs the `streaming_client.receive_stream()` function asynchronously as the status is checked. I can refactor this name to `streaming_asyncio_task` for clarity.

Reply:
👍 much clearer